pyo3-derive-backend/src/method.rs

```diff
@@ -279,10 +344,15 @@ pub fn check_arg_ty_and_optional<'a>(
     }
 }
 
-fn parse_attributes(attrs: &mut Vec<syn::Attribute>) -> syn::Result<(FnType, Vec<Argument>)> {
+fn parse_attributes(
+    attrs: &mut Vec<syn::Attribute>,
+    allow_custom_name: bool,
+) -> syn::Result<(FnType, Vec<Argument>, Option<syn::Ident>)> {
```

Comment:
The return type should probably be replaced with a named `struct`, something like:

```rust
struct ParsedAttributes {
    fn_type: FnType,
    arguments: Vec<Argument>,
    python_name: Option<syn::Ident>,
}

fn parse_attributes(
    attrs: &mut Vec<syn::Attribute>,
    allow_custom_name: bool,
) -> syn::Result<ParsedAttributes> {
```